backend/src/openarchiefbeheer/destruction/api/viewsets.py

```diff
@@ -221,18 +224,50 @@ class DestructionListViewSet(
     viewsets.GenericViewSet,
 ):
     serializer_class = DestructionListWriteSerializer
-    queryset = (
-        DestructionList.objects.all()
-        .select_related("author", "assignee")
-        .prefetch_related(
-            Prefetch(
-                "assignees",
-                queryset=DestructionListAssignee.objects.prefetch_related(
-                    "user__user_permissions"
-                ).order_by("pk"),
-            )
-        )
+    user_prefetch = User.objects.prefetch_related(
+        Prefetch(
+            "user_permissions",
+            queryset=Permission.objects.select_related("content_type").order_by(
+                "codename"
+            ),
+        ),
+        Prefetch(
+            "groups",
+            queryset=Group.objects.prefetch_related(
+                Prefetch(
+                    "permissions",
+                    queryset=Permission.objects.select_related("content_type").order_by(
+                        "codename"
+                    ),
+                )
+            ),
+        ),
+    ).annotate(
+        user_permission_codenames=ArrayAgg("user_permissions__codename", distinct=True),
+        group_permission_codenames=ArrayAgg(
+            "groups__permissions__codename", distinct=True
+        ),
     )
+    queryset = DestructionList.objects.all().prefetch_related(
+        Prefetch(
+            "author",
+            queryset=user_prefetch,
+        ),
+        Prefetch(
+            "assignee",
+            queryset=user_prefetch,
+        ),
+        Prefetch(
+            "assignees",
+            queryset=DestructionListAssignee.objects.prefetch_related(
+                Prefetch(
+                    "user",
+                    queryset=user_prefetch,
+                )
+            ).order_by("pk"),
+        ),
+    )
```

Comment:
Maybe this can be moved to a get_queryset method, since it's quite large?

Reply:
Let's see if we can refactor this along with the previous comment